internal/sync/syncer.go

```diff
@@ -233,14 +234,36 @@ func (r *resourceInfoProviderStub) IsNamespaced(_ schema.GroupKind) (bool, error
 	return false, nil
 }
 
-func (s *Syncer) sync(gitSync *v1alpha1.GitSync, targetObjs []*unstructured.Unstructured) bool {
-	// TODO: if the states match, then skip the syncing.
+func (s *Syncer) sync(gitSync *v1alpha1.GitSync, targetObjs []*unstructured.Unstructured, logger *zap.SugaredLogger) bool {
```

Comment:
can we add a comment to this function to say what the return boolean indicates? I assume error?

Reply:
Actually do you mind we do it in the PR for https://github.com/numaproj-labs/numaplane/issues/133, the `sync` method will be updated accordingly there.

Reply:
so, you're saying the signature of this method will change anyway?

Reply:
yes

Reply:
okay